pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/util/SegmentDeletionManagerTest.java

```diff
@@ -619,14 +643,32 @@ public boolean delete(URI uri, boolean forceDelete)
         _tableDirs.remove(uri.getPath() + "/");
         return true;
       }
-      // remote the segment
+      // remove the segment
       String tableName = uri.getPath().substring(0, uri.getPath().lastIndexOf("/") + 1);
       return _tableDirs.get(tableName).remove(uri.getPath());
     }
 
+    @Override
+    public boolean deleteBatch(List<URI> segmentUris, boolean forceDelete)
+        throws IOException {
+      // the expectation here is that the batch delete call is only limited to segments.
+      URI segmentURI = segmentUris.get(0);
+      String tableName = segmentURI.getPath().substring(0, segmentURI.getPath().lastIndexOf("/") + 1);
+      if (_tableDirs.containsKey(tableName)) {
+        // remove all the segments from the table directory
+        segmentUris.forEach(segmentUri -> _tableDirs.get(tableName).remove(segmentUri.getPath()));
+      }
+      // the table does not exist and we return a true;
+      return true;
+    }
+
     @Override
     public boolean exists(URI fileUri) {
-      return fileUri.getPath().endsWith("Deleted_Segments") || _tableDirs.containsKey(fileUri.getPath() + "/");
+      String uriPath = fileUri.getPath();
+      if (uriPath.endsWith("/")) {
+        uriPath = uriPath.substring(0, uriPath.lastIndexOf("/"));
+      }
+      return uriPath.endsWith("Deleted_Segments") || _tableDirs.containsKey(uriPath + "/");
```

Comment:
nit : lets make `DELETED_SEGMENTS` constant public and use it for all the occurrences in the test file